src/apps/integrations/service/integrations.py

```diff
@@ -52,23 +52,22 @@ async def create_integration(
     ) -> LorisProjects:
         match type:
             case AvailableIntegrations.LORIS:
-                return await self._create_integration(type, params)
+                return await LorisIntegrationService(
+                    params.applet_id, self.session, self.user
+                ).create_loris_integration(params)
             case _:
                 raise UnsupportedIntegrationError(type=type)
 
-    async def _create_integration(self, type, params) -> LorisProjects:
-        token = await LorisClient.login_to_loris(params.hostname, params.username, params.password)
-        projects_raw = await LorisClient.list_projects(params.hostname, token)
-        await IntegrationsCRUD(self.session).create(
+    async def retrieve_integration(self, applet_id, type) -> Integration:
+        integration_schema = await IntegrationsCRUD(self.session).retrieve(
             IntegrationsSchema(
-                applet_id=params.applet_id,
+                applet_id=applet_id,
                 type=type,
-                configuration={
-                    "hostname": params.hostname,
-                    "username": params.username,
-                    "password": params.password,
-                },
             )
         )
-        projects = list(projects_raw["Projects"].keys())
-        return LorisProjects(projects=projects)
+        existing_integration = Integration(
+            integration_type=integration_schema.type,
+            applet_id=integration_schema.applet_id,
+            configuration=integration_schema.configuration,
```

Comment:
This returns `password` as part of the response, which I think shouldn't be included.